### PR TITLE
Replace pad with cat for better performance

### PR DESCRIPTION
**Pull request by yuan-luo**
## Motivation

Using concatenation tends to be faster than padding.

image

image

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

**Review by mickqian — APPROVED** [review]

File: python/sglang/srt/models/qwen2_vl.py [hunk 1]
```diff
@@ -407,7 +407,7 @@ def forward(
         cu_seqlens = torch.repeat_interleave(
             grid_thw[:, 1] * grid_thw[:, 2], grid_thw[:, 0]
         ).cumsum(dim=0, dtype=torch.int32)
-        cu_seqlens = F.pad(cu_seqlens, (1, 0), "constant", 0)
+        cu_seqlens = torch.cat([cu_seqlens.new_zeros(1), cu_seqlens])
```
Comment (BBuf) [inline python/sglang/srt/models/qwen2_vl.py:410]: The code is:

```python
# compute cu_seqlens - move cu_seqlens to GPU and make it int32
        cu_seqlens = torch.cat(
            [
                torch.tensor([0], device=x.device, dtype=torch.int32),
                (grid_thw[:, 0] * grid_thw[:, 1] * grid_thw[:, 2])
                .cumsum(dim=0)
                .to(device=x.device, dtype=torch.int32),
            ]
        )
        cu_seqlens = torch.cat([cu_seqlens.new_zeros(1), cu_seqlens])
```

It'll cause `cu_seqlens = [0, 0, seq_len1, seq_len1+seq_len2, ...]` , is a bug?

Comment (yuan-luo) [reply python/sglang/srt/models/qwen2_vl.py:410]: The original version `F.pad(cu_seqlens, (1, 0), "constant", 0)` also prepend a 0, the PR is just a pure functional replacement, doesn't change the logic.

**Review by BBuf — APPROVED** [review]
lgtm

**Comment by fzyzcjy** [comment]
qq: curious is it only useful to reduce cpu time, or it is also helpful for gpu time
